### PR TITLE
edited hash function, main menu buffer, BST

### PR DESCRIPTION
**Pull request by kevwjin**

### CODE REVIEW

**Review by lance60 — CHANGES_REQUESTED** [review]
hash.h is causing read access violation problems when reading more than 39 data from db.csv. Search by address still doesnt work for certain address keys that exists in the hash table (Issue probably lies in the new hash function).

File: hash.h [hunk 1]
```diff
@@ -173,7 +211,7 @@ int Hash<T>::getAverageListSize()
 }
 
 template<class T>
-int Hash<T>::getNumCollisions() 
+int Hash<T>::getNumCollisions()
```
Comment (Radical-Egg) [inline hash.h:214]: This did not need to be changed, please refer to master branch

File: hash.h [hunk 2]
```diff
@@ -121,7 +159,7 @@ class Hash
 };
 
 template<class T>
-int Hash<T>::getLongestListSize() 
+int Hash<T>::getLongestListSize()
```
Comment (Radical-Egg) [inline hash.h:162]: this did not need to be changed please refer to master branch

File: hash.h [hunk 3]
```diff
@@ -203,7 +241,11 @@ template<class T>
 int Hash<T>::hashFunction(std::string key)
 {
     // get int value of key
-    int hash_value = key.length();
+    int charSum = 0;
```
Comment (Radical-Egg) [inline hash.h:244]: line 244-248 change back to previous (line 206) 
int hash_value = key.length()

**Pull request closed without merging**